contracts/SolverTrampoline.sol

```diff
@@ -3,5 +3,34 @@ pragma solidity ^0.8.17;
 
 import { Authentication, Settlement } from "./CoWProtocol.sol";
 
+/// @title CoW Protocol Solver Trampoline
+/// @author CoW Protocol Developers
+/// @dev A solver trampoline contract that allows permissionless execution of
+/// signed authorized settlements. This can be used to allow relayers to execute
+/// settlements on behalf solvers without being a registered solver themselves.
 contract SolverTrampoline {
+    /// @dev The domain separator for signing EIP-712 settlements.
+    bytes32 public immutable domainSeparator;
+
+    constructor() {
+        domainSeparator = keccak256(abi.encode(
```

Comment:
I wonder if the "issues" we faced after the POW<>POS merge (our contract now existing on a chain where domainSeparator does not agree with the chainId) would make us reconsider hardcoding this value (rather than computing it on each run)?

Reply:
Implemented in https://github.com/cowprotocol/solver-trampoline/pull/9. It adds 200 gas per settlement.